pkg/services/ngalert/notifier/receiver_svc_test.go

```diff
@@ -0,0 +1,208 @@
+	})
+
+	t.Run("service decrypts receiver groups with permission", func(t *testing.T) {
+		sut := createReceiverServiceSut(t, secretsService)
+		sut.ac = ac
+
+		q := rQuery(1, "slack receiver")
+		q.Decrypt = true
+		Receivers, err := sut.GetReceivers(context.Background(), q, &user.SignedInUser{
+			OrgID: 1,
+			Permissions: map[int64]map[string][]string{
+				1: {accesscontrol.ActionAlertingProvisioningReadSecrets: nil},
+			},
+		})
+		require.NoError(t, err)
+
+		require.Len(t, Receivers, 1)
+
+		rGroup := Receivers[0]
+		require.Equal(t, "slack receiver", rGroup.Name)
+		require.Len(t, rGroup.GrafanaManagedReceivers, 1)
+
+		grafanaReceiver := rGroup.GrafanaManagedReceivers[0]
+		require.Equal(t, "UID2", grafanaReceiver.UID)
+
+		settings, err := simplejson.NewJson([]byte(grafanaReceiver.Settings))
+		require.NoError(t, err)
+		require.Equal(t, "secure url", settings.Get("url").MustString())
+	})
+}
+
+func createReceiverServiceSut(t *testing.T, encryptSvc secrets.Service) *ReceiverService {
+	cfg := createEncryptedConfig(t, encryptSvc)
+	store := fakes.NewFakeAlertmanagerConfigStore(cfg)
+	xact := newNopTransactionManager()
+	provisioningStore := fakes.NewFakeProvisioningStore()
+
+	return &ReceiverService{
+		actest.FakeAccessControl{},
+		provisioningStore,
+		store,
+		encryptSvc,
+		xact,
+		log.NewNopLogger(),
+	}
+}
+
+func createEncryptedConfig(t *testing.T, secretService secrets.Service) string {
+	c := &definitions.PostableUserConfig{}
+	err := json.Unmarshal([]byte(defaultAlertmanagerConfigJSON), c)
+	require.NoError(t, err)
+	err = EncryptReceiverConfigs(c.AlertmanagerConfig.Receivers, func(ctx context.Context, payload []byte) ([]byte, error) {
+		return secretService.Encrypt(ctx, payload, secrets.WithoutScope())
+	})
+	require.NoError(t, err)
+	bytes, err := json.Marshal(c)
+	require.NoError(t, err)
+	return string(bytes)
+}
+
+func rQuery(orgID int64, names ...string) models.GetReceiversQuery {
+	return models.GetReceiversQuery{
+		OrgID: orgID,
+		Names: names,
+	}
+}
+
+const defaultAlertmanagerConfigJSON = `
+{
+	"template_files": null,
+	"alertmanager_config": {
+		"route": {
+			"receiver": "grafana-default-email",
+			"group_by": [
+				"..."
+			],
+			"routes": [{
+				"receiver": "grafana-default-email",
+				"object_matchers": [["a", "=", "b"]]
+			}]
+		},
+		"templates": null,
+		"receivers": [{
+			"name": "grafana-default-email",
+			"grafana_managed_receiver_configs": [{
+				"uid": "UID1",
+				"name": "grafana-default-email",
+				"type": "email",
+				"disableResolveMessage": false,
+				"settings": {
+					"addresses": "\u003cexample@email.com\u003e"
+				},
+				"secureFields": {}
+			}]
+		}, {
+			"name": "slack receiver",
+			"grafana_managed_receiver_configs": [{
+				"uid": "UID2",
+				"name": "slack receiver",
+				"type": "slack",
+				"disableResolveMessage": false,
+				"settings": {},
+				"secureSettings": {"url":"secure url"}
+			}]
+		}]
+	}
+}
+`
```

Comment:
for context: one of the reasons to implement the configStore abstraction was to avoid big JSON blobs in the unit tests and create structs in the fake. I hope we'll get rid of this later.